src/src/beneficiaries/beneficiaries.service.ts

```diff
@@ -297,10 +294,278 @@ if(!result){
   remove(id: number) {
     // return this.hasuraService.delete(this.table, { id: +id });
   }
+  
  public async statusUpdate(req:any){
-    console.log("req",req)
   return await this.hasuraService.update(req.id, 'beneficiaries',req,this.returnFields,[...this.returnFields,"id"]);
 
   }
-}
 
+  public async registerBeneficiary(body, request) {
+    const user=await this.userService.ipUserInfo(request);
+    const password = body.mobile;
+    let username = body.first_name;
+    username += `_${body.mobile}`;
+    const data_to_create_user = {
+      enabled: 'true',
+      firstName: body.first_name,
+      username: username.toLowerCase(),
+      credentials: [
+        {
+          type: 'password',
+          value: password,
+          temporary: false,
+        },
+      ],
+      groups: ['beneficiaries'],
+    };
+
+    const data = {
+      username: 'admin',
+      client_id: 'admin-cli',
+      grant_type: 'client_credentials',
+      password: this.configService.get<string>('KEYCLOAK_ADMIN_PASSWORD'),
+      client_secret: this.configService.get<string>('KEYCLOAK_ADMIN_CLI_CLIENT_SECRET'),
+    };
+    const adminResultData = await this.keycloakService.getAdminKeycloakToken(data, 'master');
+    
+    if (adminResultData?.access_token) {
+      let url = `${this.configService.get<string>('KEYCLOAK_URL')}/admin/realms/eg-sso/users`;
+      let data = data_to_create_user;
+
+      try {
```

Comment:
can we create and use same method to create prerak, ag in Keycloak?

create and use common method for same? 
@parthgandhitekdi @suraj-tekdi

Reply:
Created common method in keycloak service file to create user on keycloak.